qiskit/transpiler/passes/scheduling/padding.py

```diff
@@ -125,27 +132,46 @@ def run(self, dag: DAGCircuit):
 
         # Add delays until the end of circuit.
         for bit in new_dag.qubits:
-            idle_time = circuit_duration - idle_after[bit]
-            node = new_dag.output_map[bit]
-            prev_node = next(new_dag.predecessors(node))
-            if idle_time > 0:
+            if circuit_duration - idle_after[bit] > 0:
+                node = new_dag.output_map[bit]
+                prev_node = next(new_dag.predecessors(node))
                 self._pad(
                     dag=new_dag,
                     qubit=bit,
-                    time_interval=idle_time,
+                    t_start=idle_after[bit],
+                    t_end=circuit_duration,
                     next_node=node,
                     prev_node=prev_node,
                 )
 
         new_dag.duration = circuit_duration
 
+        # Invalidate old schedule information since delays are filled with sequence.
+        del self.property_set["node_start_time"]
```

Comment:
I wonder if "invalidating" is actually the right thing to do. These Padding passes do not actually change absolute timing of the circuit, right? i.e. if node N had `t0=1000`, then after PadDelay or DynamicalDecoupling, N will still have `t0=1000`.

But I think you do this because you are concerned that calling PadDelay or DynamicalDecoupling a second time will be invalid because we need new t0 information for the new sequence gates. So is it more correct to say that this information is "incomplete", rather than "invalid"? If so, then I think the second call to the pass should automatically fail because it will encounter some nodes with no t0 information.

Reply:
Probably you are right. We can update node start time so that successor passes can still utilize the scheduled information, rather than invalidating it, with small overhead. Please check new logic and test implemented in https://github.com/Qiskit/qiskit-terra/pull/7760/commits/b271472642a254c35b05df7ed340c1bdb3f54ad3

Now we can call DD pass twice like
```python
pm = PassManager(
    [
        ALAPSchedule(self.durations),
        DynamicalDecoupling(self.durations, dd_sequence, qubits=[0]),
        DynamicalDecoupling(self.durations, dd_sequence, qubits=[1]),
    ]
)
```
the updated scheduled info can be directly consumed by the timeline drawer in follow-up PR.

Reply:
so calling the DD pass a second time has no effect here? can you explain why?

Reply:
It has effect because I set different qubit. The first DD inserts the sequence for q0 and second DD is for q1 (I think this is not practical from viewpoint of performance but a good case for test of logic). The trick is easy. The base padding pass always ignore the delay instructions when it constructs circuits, i.e. delay = idle time = the pass can pad the sequence. 
https://github.com/nkanazawa1989/qiskit-terra/blob/19d235b8f1d6ad6c54741936367037a058f9b065/qiskit/transpiler/passes/scheduling/base_padding.py#L99-L104

In the above configuration, the first DD inserts delay for the idle time on q1 wire (this is due to backward compatibility), but these delays are removed in the line above, and thus the second DD can overrides the delays with the sequence.